plugins/api-docs/src/alpha.tsx

```diff
@@ -0,0 +1,205 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+
+import {
+  createApiExtension,
+  createApiFactory,
+  createNavItemExtension,
+  createPageExtension,
+  createPlugin,
+  createSchemaFromZod,
+} from '@backstage/frontend-plugin-api';
+import {
+  compatWrapper,
+  convertLegacyRouteRef,
+} from '@backstage/core-compat-api';
+import { ApiEntity } from '@backstage/catalog-model';
+
+import { defaultDefinitionWidgets } from './components/ApiDefinitionCard';
+import { rootRoute, registerComponentRouteRef } from './routes';
+import { apiDocsConfigRef } from './config';
+import {
+  createEntityCardExtension,
+  createEntityContentExtension,
+} from '@backstage/plugin-catalog-react/alpha';
+import { Grid } from '@material-ui/core';
+import { useApp } from '@backstage/core-plugin-api';
+
+function ApiIcon() {
+  const app = useApp();
+  const KindApiSystemIcon = app.getSystemIcon('kind:api')!;
+  return <KindApiSystemIcon />;
+}
+
+const ApiDocsNavItem = createNavItemExtension({
+  title: 'APIs',
+  routeRef: convertLegacyRouteRef(rootRoute),
+  icon: () => compatWrapper(<ApiIcon />),
+});
+
+const ApiDocsConfigApi = createApiExtension({
+  factory: createApiFactory({
+    api: apiDocsConfigRef,
+    deps: {},
+    factory: () => {
+      const definitionWidgets = defaultDefinitionWidgets();
+      return {
+        getApiDefinitionWidget: (apiEntity: ApiEntity) => {
+          return definitionWidgets.find(d => d.type === apiEntity.spec.type);
+        },
+      };
+    },
+  }),
+});
+
+const ApiDocsExplorerPage = createPageExtension({
+  defaultPath: '/api-docs',
+  routeRef: convertLegacyRouteRef(rootRoute),
+  // Mapping DefaultApiExplorerPageProps to config
+  configSchema: createSchemaFromZod(z =>
+    z.object({
+      path: z.string().default('/api-docs'),
+      initiallySelectedFilter: z.enum(['owned', 'starred', 'all']).optional(),
+      // Ommiting columns and actions for now as their types are too complex to map to zod
+    }),
+  ),
+  loader: ({ config }) =>
+    import('./components/ApiExplorerPage').then(m =>
+      compatWrapper(
+        <m.ApiExplorerIndexPage
+          initiallySelectedFilter={config.initiallySelectedFilter}
+        />,
+      ),
+    ),
+});
+
+const ApiDocsHasApisEntityCard = createEntityCardExtension({
+  name: 'has-apis',
+  // we are skipping variants, see: https://github.com/backstage/backstage/pull/22619#discussion_r1477333252
+  // and columns are too complex to map to zod
+  loader: () => import('./components/ApisCards').then(m => <m.HasApisCard />),
```

Comment:
Should this one have a `compatWrapper` too?

Reply:
Oops, fixed.